test/engine-shared-conditions.test.js

```diff
@@ -0,0 +1,106 @@
+'use strict'
+
+import sinon from 'sinon'
+import engineFactory from '../src/index'
+
+describe('Engine: shared conditions', () => {
```

Comment:
Consider adding more test cases:
- Using multiple shared conditions
- Trying to use a shared condition that doesn't exist